app/code/Magento/Developer/Model/TemplateEngine/Plugin/DebugHints.php

```diff
@@ -94,15 +124,37 @@ public function afterCreate(
         $storeCode = $this->storeManager->getStore()->getCode();
         if ($this->scopeConfig->getValue($this->debugHintsPath, ScopeInterface::SCOPE_STORE, $storeCode)
             && $this->devHelper->isDevAllowed()) {
-            $showBlockHints = $this->scopeConfig->getValue(
-                self::XML_PATH_DEBUG_TEMPLATE_HINTS_BLOCKS,
+            $debugHintsWithParam = $this->scopeConfig->getValue(
+                $this->debugHintsWithParam,
+                ScopeInterface::SCOPE_STORE,
+                $storeCode
+            );
+            $debugHintsParameter = $this->scopeConfig->getValue(
+                $this->debugHintsParameter,
                 ScopeInterface::SCOPE_STORE,
                 $storeCode
             );
-            return $this->debugHintsFactory->create([
-                'subject' => $invocationResult,
-                'showBlockHints' => $showBlockHints,
-            ]);
+            $debugHintsParameterInUrl = $this->http->getParam('templatehints');
+
+            $showHints = false;
+            if (!$debugHintsWithParam) {
+                $showHints = true;
+            }
+            if ($debugHintsWithParam && $debugHintsParameter == $debugHintsParameterInUrl) {
+                $showHints = true;
+            }
```

Comment:
Overcomplicated implementation with unnecessarily high cyclomatic complexity.

Equivalent implementation:
```php
$showHints = !$debugHintsWithParam || $debugHintsParameter == $debugHintsParameterInUrl;
```

Reply:
I wrote something similar to that at first and the original CI review process told me to change it. But feel free to do whatever needs to be done. I still think this is a useful feature.

Reply:
Just wanted to leave a comment in case someone stumbles upon it.
I am not part of the Magento team (not anymore) and no action is needed on your part :)

Reply:
The feature is indeed useful. Thank you so much for contributing it!
Being the original author of that code, I'm just being sensitive to increase of its complexity :)

Reply:
@sshymko,

> Just wanted to leave a comment in case someone stumbles upon it.

How is your comment ever useful to anybody? Please provide a use case.

Thanks for paying attention to code quality though. And also let me inform you that we are accepting pull requests for more than a year already ;) This is the most natural way of constructive collaboration.

Reply:
I'll try to find time to learn the guidelines and submit the PR.
Until then the comment serves as a reminder to do that.